File: 199-binary-tree-right-side-view/binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> rightSideView(TreeNode* root) {
         vector<int> ans;
        if (root == NULL)
            return ans;

        queue<TreeNode*> pq;
        pq.push(root);
        while (!pq.empty()) {
            int levelsize = pq.size();
            for (int i = 0; i < levelsize; i++) {
                TreeNode* node = pq.front();
                pq.pop();
                if (node->left)
                    pq.push(node->left);
                if (node->right)
                    pq.push(node->right);
                if(i == levelsize-1)ans.push_back(node->val);
            }
           
        }
        return ans;
    }
};

File: 199-binary-tree-right-side-view/binary-tree-right-side-view.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var rightSideView = function(root) {
    let result =[];
    if(!root)return [];
    const queue = [root];
    while(queue.length){
        let size = queue.length;
        while(size){
            let currentNode = queue.shift();
            if(currentNode.left)queue.push(currentNode.left);
            if(currentNode.right)queue.push(currentNode.right);
            if(size==1)result.push(currentNode.val);
            size--;
        }
    }
    return result;
    
};